aws-storage-s3/src/androidTest/java/com/amplifyframework/storage/s3/AWSS3StorageDownloadAccessLevelTest.java

```diff
@@ -0,0 +1,239 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amplifyframework.storage.s3;
+
+import com.amplifyframework.storage.StorageAccessLevel;
+import com.amplifyframework.storage.StorageException;
+import com.amplifyframework.storage.options.StorageDownloadFileOptions;
+import com.amplifyframework.testutils.FileAssert;
+import com.amplifyframework.testutils.RandomTempFile;
+import com.amplifyframework.testutils.SynchronousAWSMobileClient;
+
+import org.junit.AfterClass;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static org.junit.Assert.assertTrue;
+
+/**
+ * Instrumentation test to confirm that Storage Download behaves
+ * correctly with regards to the provided storage access level.
+ */
+public final class AWSS3StorageDownloadAccessLevelTest extends StorageInstrumentationTestBase {
+
+    private static String userNameOne;
+    private static String userNameTwo;
+    private static Map<String, String> identityIds = new HashMap<>();
+
+    private static final String UPLOAD_NAME = "upload-test-" + System.currentTimeMillis();
+    private static final long UPLOAD_SIZE = 100L;
+    private static File uploadFile;
+
+    private final String destination = "download-test-" + System.currentTimeMillis();
+    private File downloadFile;
+
+    /**
+     * Upload the required resources in cloud prior to running the tests.
+     * @throws IOException if there is a problem while creating the files
+     * @throws StorageException if there is a problem with uploading the files
+     * @throws SynchronousAWSMobileClient.MobileClientException from failure to sign in with
+     *         Cognito User Pools or obtain an identity ID
+     */
+    @BeforeClass
+    public static void setUpBeforeClass() throws IOException,
+            StorageException,
+            SynchronousAWSMobileClient.MobileClientException {
```

Comment:
You might want to call `cleanup()` inside of here, too, since `AfterClass` is not guaranteed to run. (If the process crashes during a test, AfterClass won't ever run, for example.)